Source/WebCore/page/FocusController.cpp

```diff
@@ -396,7 +396,8 @@ Frame& FocusController::focusedOrMainFrame() const
 {
     if (Frame* frame = focusedFrame())
         return *frame;
-    return m_page.mainFrame();
+
+    return *dynamicDowncast<LocalFrame>(m_page.mainFrame());
```

Comment:
null check.

Reply:
we're returning a reference here, so maybe adding an assert is the way to go for this